### ISSUE TITLE
Error syncing the same subgraph using two or more index-node-x 

### ISSUE THREAD

**Issue by Aminechakr** [graphprotocol/graph-node#3318]
**report a bug**

**What is the current behavior?**
I've set up the recommended infrastructure for Graphnode, by splitting the component into a query-node & index-node.

When scaling up the index-node I'm unable to sync the same subgraph with both index-node-1 & index-node-2
What i'm doing wrong ? been using the following helm charts : 

- https://artifacthub.io/packages/helm/vulcanlink/graphprotocol-indexnode

```
index-node-0                                               1/1     Running       0          66m
index-node-1                                               1/1     Running       0          66m
index-node-2                                               1/1     Running       2          67m
```
`index-node-0` is running as a blockIngestor from Rinkeby.
`index-node-1` deployed first subgraph on it.
`index-node-2` using the same IPFS hash file for the subgraph that was deployed on `index-node-1` and deployed it under the second `index-node-2`.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

Deploy the following chart in K8S : https://artifacthub.io/packages/helm/vulcanlink/graphprotocol-indexnode
Deploy custom subgraph  in first statefulSet pod.
Deploy same custom subgraph in second statefulSet pod.
`index-node1` is syncing correctly while `index-node-2`  is stuck at : 

```
Mar 04 02:33:00.691 INFO Set subgraph start block, block_hash: Some(0xe6f37e2b1a92ff042d23a75e0b9ebde7fc7ad8274458c526102a9bcea72be674), block_number: Some(1795515), subgraph_id: QmUjcRpzgYiB6ZMBw9wczcjnYh4MVTc41vqQMVkhYf4TZa, component: SubgraphRegistrar
Mar 04 02:33:00.691 INFO Graft base, block: None, base: None, subgraph_id: QmUjcRpzgYiB6ZMBw9wczcjnYh4MVTc41vqQMVkhYf4TZa, component: SubgraphRegistrar
```

**What is the expected behavior?**
Both index-node-1 & index-node-2 should be syncing the same subgraph.

**Comment by Aminechakr** [comment]
What i noticed is that i can scale only vertically, and not horizontally. any insight ?

**Comment by Aminechakr** [comment]
Tried discovering `@graphprotocol/indexer-cli@0.19.0 ` but the build is failing.

**Comment by azf20** [comment]
hi @Aminechakr - each subgraph must be indexed by just one graph-node at a time, scaling horizontally as you describe is not possible. Are you encountering issues which led you to try this approach?

**Comment by yohanelly95** [comment]
@azf20 can one graph node write to another graph nodes postgres db? 

So I am trying to run two graph nodes, but since I am using skale some events are not indexed properly.(missing events when there is more than 1 event emitted from contract). 

Node 1: full subgraph with all events + ipfs 1 + postgres 1
Node 2: partial subgraph + ipfs 2 + postgres 1. 

I cant seem to write to the 1st graph nodes db, any pointers here?

**Comment by Aminechakr** [comment]
Hi @azf20, 

I'm trying to build an EVM NFT Indexer on the Ethereum chain based on The Graph protocol indexer, the service should be highly scalable and available and will be indexing a LOT of entities. Those entities need to be queryable from specific endpoints, those endpoints should also be scalable since a lot of users will be pulling information from FE side.

I made a proxy and query service running with HPA enabled, that works fine.

Facing trouble with the indexers only, actually, I have deployed 3 subgraphs in 3 indexer nodes, a fourth node is pulling only data from my Ethereum archive node (block ingestor job only).

The issue I'm facing now is that since scalability is hardly managed on the indexers ( I need to redeploy my subgraph each time I change the capacity type of my nodes which is not great because I lost all the indexed data that was done by the subgraph. I'm still looking for a way to better manage this, and also see if I can backup and restore the indexed data so it not be lost forever.
